x/harvest/keeper/borrow.go

```diff
@@ -0,0 +1,138 @@
+}
+
+// ValidateBorrow validates a borrow request against borrower and protocol requirements
+func (k Keeper) ValidateBorrow(ctx sdk.Context, borrower sdk.AccAddress, amount sdk.Coin) error {
+	if err := k.validateBorrowUser(ctx, borrower, amount); err != nil {
+		return err
+	}
+
+	if err := k.validateBorrowSystem(ctx, borrower, amount); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (k Keeper) validateBorrowUser(ctx sdk.Context, borrower sdk.AccAddress, amount sdk.Coin) error {
+	var borrowMoneyMarket types.MoneyMarket
+	var borrowAssetPrice sdk.Dec
+	var borrowUSDValue sdk.Dec
+	if amount.Denom == USDX {
+		borrowUSDValue = sdk.NewDecFromInt(amount.Amount).Quo(sdk.NewDec(int64(math.Pow(10, 6)))) // TODO: include conversion_factor in params
+	} else {
+		locBorrowMoneyMarket, found := k.GetMoneyMarket(ctx, amount.Denom)
+		if !found {
+			return sdkerrors.Wrapf(types.ErrMarketNotFound, "no market found for denom %s", amount.Denom)
+		}
+		locBorrowAssetPrice, err := k.pricefeedKeeper.GetCurrentPrice(ctx, locBorrowMoneyMarket.SpotMarketID)
+		if err != nil {
+			return sdkerrors.Wrapf(types.ErrPriceNotFound, "1: no price found for market %s", locBorrowMoneyMarket.SpotMarketID)
+		}
+		borrowMoneyMarket = locBorrowMoneyMarket
+		borrowAssetPrice = locBorrowAssetPrice.Price
+		borrowUSDValue = sdk.NewDecFromInt(amount.Amount).Mul(borrowAssetPrice)
+	}
+
+	// Get the user's deposits
+	deposits := k.GetDepositsByUser(ctx, borrower)
+	if len(deposits) == 0 {
+		return sdkerrors.Wrapf(types.ErrDepositsNotFound, "no deposits found for %s", borrower)
+	}
+
+	// Get the total value of the user's deposits
+	totalValueDeposits := sdk.ZeroDec()
+	for _, deposit := range deposits {
+		if deposit.Amount.Denom == USDX {
+			totalValueDeposits = totalValueDeposits.Add(sdk.NewDecFromInt(deposit.Amount.Amount))
+		} else {
+			depositMoneyMarket, found := k.GetMoneyMarket(ctx, deposit.Amount.Denom)
+			if !found {
+				return sdkerrors.Wrapf(types.ErrMarketNotFound, "no market found for denom %s", deposit.Amount.Denom)
+			}
+			depositAssetPrice, err := k.pricefeedKeeper.GetCurrentPrice(ctx, depositMoneyMarket.SpotMarketID)
+			if err != nil {
+				return sdkerrors.Wrapf(types.ErrPriceNotFound, "2: no price found for market %s", depositMoneyMarket)
+			}
+			depositUSDValue := sdk.NewDecFromInt(deposit.Amount.Amount).Mul(depositAssetPrice.Price)
+			totalValueDeposits = totalValueDeposits.Add(depositUSDValue)
+		}
+	}
+
+	// Get the total value of the user's borrows
+	borrows := k.GetBorrowsByUser(ctx, borrower)
+	totalValueBorrows := sdk.ZeroDec()
+	for _, borrow := range borrows {
+		if borrow.Amount.Denom == USDX {
+			totalValueBorrows = totalValueBorrows.Add(sdk.NewDecFromInt(borrow.Amount.Amount))
+		} else {
+			borrowAssetPrice, err := k.pricefeedKeeper.GetCurrentPrice(ctx, borrowMoneyMarket.SpotMarketID)
+			if err != nil {
+				return err
+			}
+			borrowUSDValue := sdk.NewDecFromInt(borrow.Amount.Amount).Mul(borrowAssetPrice.Price)
```

Comment:
This will overwrite the previous value of borrowUSDValue, make the comparison on line 126 invalid.

Reply:
Good catch, resolved in https://github.com/Kava-Labs/kava/pull/702/commits/5428519008577591be427331d9501c142a1e77c8